### ISSUE TITLE
When the very first character is written on an input, it is not binded with $scope, and then $invalid flag takes a worng value

### ISSUE THREAD

**Issue by atmjesus** [nelsonomuto/angular-ui-form-validation#60]
Hello!

First of all, I would like to thank you for this great and useful library.

I have detected an issue when an angular-ui-form-validation directive is assigned to a form input field:

1) I code a sample html form, dumping $scope.name and $invalid variables: 
![code1](https://cloud.githubusercontent.com/assets/10669250/5874319/81ed2e60-a300-11e4-9a94-8a541dab5601.png)

As you could see below, the result is correct, since $scope.name='' and $invalid=false:
![sampleform](https://cloud.githubusercontent.com/assets/10669250/5874327/8a32be1e-a300-11e4-9a19-6f531f61e09f.png)

2) <b>Here comes the issue</b>: When the very first character is written on the input field, it is expected that $scope.name='a' and $invalid=false, however, the input is not binded with its corresponding $scope variable, and then $invalid is set to true:
![sampleform2](https://cloud.githubusercontent.com/assets/10669250/5876631/1042adee-a319-11e4-83e1-c6156a35f9fd.png)

3) Then, I insert the second character. After this, the variable is binded and the validation process works fine
![sampleform3](https://cloud.githubusercontent.com/assets/10669250/5876662/5a2ca748-a319-11e4-9e20-c35c404e094f.png)
If I delete the just inserted character, then the directive is still working:
![sampleform4](https://cloud.githubusercontent.com/assets/10669250/5876668/5ef5e320-a319-11e4-93c9-349e67f37f4e.png)

4) After step 3, if I delete the last character and the input lost the focus, the validation directive works fine
![sampleform5](https://cloud.githubusercontent.com/assets/10669250/5876696/8ad8a766-a319-11e4-945f-134b5872c04a.png)

5) Then, if I put another random character, the validation works
![sampleform6](https://cloud.githubusercontent.com/assets/10669250/5876740/f7bd6448-a319-11e4-80ed-7f238d8b7fc3.png)

Any ideas/solutions? Thank you very much

**Comment by nelsonomuto** [comment]
I just ran a test for this in my debugger and I don't see the issue. Everything is firing as expected and the binding is accurate.

In the validator, you can inspect the form model controller, it is passed in as the 6th parameter, it corresponds to `sampleForm.namefield` (the form controller model property)
     
    function (errorMessageElement, val, attr, $element, model, ngModelController, $scope, rawAttr) {

**Comment by nelsonomuto** [comment]
As far as the UI bound value `{{sampleForm.username.$valid}}`, you are correct to observe that it does not update on the first run of the validation. However the value is accurately updated in the validator function corresponding value `ngModelController.$valid`. Not sure why the binding is not kicked off initially on the first run. But I will keep testing, good catch.

**Comment by atmjesus** [comment]
Thank you very much. I'll wait for a resolution. I cannot try to solve the bug since the source code is minified. 

PD. By the way, something similar happens when CTRL-C/CTRL-V is done on the input.

**Comment by atmjesus** [comment]
I just realize that this bug appears when using AngularJS>=1.3.0. With AngularJS 1.2.x the custom validation library works fine.

**Comment by nelsonomuto** [comment]
I am looking into it currently.

**Comment by nirmaladdon** [comment]
Have you checked,i am also facing this problem

**Comment by krunalshah68** [comment]
Yep faced same issue today. Model is not updating on first time input and unfortunately form gets invalidated. This causes major issue in form  validation. Hope it gets fixed ASAP.
P.S: I'm using custom validation  created in my custom module.
